### PR TITLE
Feature: Ghost movement 2

### LINKED ISSUE #4: Feature: Ghost Movement
Criteria:
- Ghost needs to be able to move forward-back-left-right (WASD)
- Ghost needs to be able to move up-down (controls subject to change)

### PR DESCRIPTION
**Pull request by hassanmohamed2003**
## Description
Added the ghost movement, ability to fly and move.

## Setting up testing environment
1. In the project window navigate to Project/Assets/Scenes and open the GhostMovement scene.
2. Press Play.

#### Testing Movement system
You are able to move forward, backwards, left and right with AWSD. You can do up with 'Space' and down with 'Ctrl'. Player movement is dependent on way the camera is facing.

Criteria:
- [x] Able to move and fly
- [x] When rotating the camera the player movement changes (looking up will make you move upwards, looking down downwards etc.)

## Checklist
_These checks should always be true for your pull request_
- [x] My pull request is for one story/feature.
- [x] Every individual commit in this pull request is logical.
- [x] All code, documentation, commits and player seen texts are in English.
- [x] I have deleted unused / unnecessary code.
- [x] If my edits need a change in documentation, then I have updated the documentation.
- [x] All code is in correspondence with the code conventions.

### CODE REVIEW

File: Assets/Scripts/Ghost/ThirdPersonController.cs [hunk 1]
```diff
@@ -15,11 +17,9 @@ namespace StarterAssets
     public class ThirdPersonController : MonoBehaviour
     {
         [Header("Player")]
-        [Tooltip("Move speed of the character in m/s")]
         public float MoveSpeed = 2.0f;
```
Comment (KiwiLT) [inline Assets/Scripts/Ghost/ThirdPersonController.cs:20]: Too much public variables. Check each variable if you really need them to be public

File: Assets/StarterAssets/InputSystem/StarterAssetsInputs.cs [hunk 2]
```diff
@@ -51,6 +63,16 @@ public void MoveInput(Vector2 newMoveDirection)
 			move = newMoveDirection;
 		} 
 
+		public void FlyInput(float newFlyDirection)
+		{
+			fly = newFlyDirection;
+		}
+
+		public void FlyingInput(bool newFlyState)
```
Comment (KiwiLT) [inline Assets/StarterAssets/InputSystem/StarterAssetsInputs.cs]: Hassan: I don't use this function

File: Assets/StarterAssets/InputSystem/StarterAssetsInputs.cs [hunk 3]
```diff
@@ -12,6 +12,8 @@ public class StarterAssetsInputs : MonoBehaviour
 		public Vector2 look;
 		public bool jump;
 		public bool sprint;
+		public float fly;
+		public bool flying;
```
Comment (KiwiLT) [inline Assets/StarterAssets/InputSystem/StarterAssetsInputs.cs]: Hassan: I don't use this variable

File: Assets/StarterAssets/InputSystem/StarterAssetsInputs.cs [hunk 4]
```diff
@@ -12,6 +12,8 @@ public class StarterAssetsInputs : MonoBehaviour
...
 
 		[Header("Movement Settings")]
 		public bool analogMovement;
```
Comment (KiwiLT) [inline Assets/StarterAssets/InputSystem/StarterAssetsInputs.cs:19]: these functions can be private

File: Assets/StarterAssets/InputSystem/StarterAssetsInputs.cs
Comment (KiwiLT) [inline Assets/StarterAssets/InputSystem/StarterAssetsInputs.cs:1]: Access modifiers: Most functions here can be private

File: Assets/Scripts/Ghost/ThirdPersonController.cs
Comment (KiwiLT) [inline Assets/Scripts/Ghost/ThirdPersonController.cs:1]: Access Modifiers: Most fields can be private here with SerializeField

**Review by KiwiLT — DISMISSED** [review]
Lots of comments of Unity, maybe you can add some more comments yourself. Biggest problem is the Access Modifiers, please use more private fields

File: Assets/Scripts/Ghost/ThirdPersonController.cs
Comment (hassanmohamed2003) [reply Assets/Scripts/Ghost/ThirdPersonController.cs:1]: changed the ones that could be changed, some had references

File: Assets/StarterAssets/InputSystem/StarterAssetsInputs.cs (hunk 3 above)
Comment (hassanmohamed2003) [reply Assets/StarterAssets/InputSystem/StarterAssetsInputs.cs]: fixed

File: Assets/StarterAssets/InputSystem/StarterAssetsInputs.cs (hunk 2 above)
Comment (hassanmohamed2003) [reply Assets/StarterAssets/InputSystem/StarterAssetsInputs.cs]: removed

File: Assets/Scripts/Ghost/ThirdPersonController.cs [hunk 5]
```diff
@@ -1,4 +1,6 @@
-﻿ using UnityEngine;
+﻿using TMPro;
+using Unity.VisualScripting;
```
Comment (Jeroenvn) [inline Assets/Scripts/Ghost/ThirdPersonController.cs]: Unused imports

File: Assets/Scripts/Ghost/ThirdPersonController.cs [hunk 6]
```diff
@@ -15,65 +17,55 @@ namespace StarterAssets
...
+        [SerializeField] private float CameraAngleOverride = 0.0f;
 
         [Tooltip("For locking the camera position on all axis")]
-        public bool LockCameraPosition = false;
+        [SerializeField]private bool LockCameraPosition = false;
+
+        [Header("Flying")]
+        public float minHeight;
+        public float maxHeight;
+        public float currentHeight;
```
Comment (Jeroenvn) [inline Assets/Scripts/Ghost/ThirdPersonController.cs]: public variables should be pascal case

**Review by Jeroenvn — CHANGES_REQUESTED** [review]
The movement works in game but there are some small code conventions that should be fixed.

**Comment by Jeroenvn** [comment]
Also, there is a missing script on the player armature

**Review by Jeroenvn — APPROVED** [review]
Looks Good!
[image](https://tenor.com/8CtX.gif)

**Pull request merged**